### PR TITLE
Refactor DB interface

### PR DESCRIPTION
**Pull request by johnugeorge**
DB interface is refactored so that backend DB is pluggable

Related: #498

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/760)

### CODE REVIEW

**Comment by johnugeorge** [comment]
For new DB support,  add corresponding implementation in `pkg/db/v1alpha3` and use `DB_NAME` to point at it.

File: cmd/manager/v1alpha3/main.go [hunk 1]
```diff
@@ -194,8 +196,11 @@ func (s *server) Check(ctx context.Context, in *health_pb.HealthCheckRequest) (*
 func main() {
 	flag.Parse()
 	var err error
-
-	dbIf, err = kdb.New()
+	dbName := os.Getenv("DB_NAME")
```
Comment (hougangliu) [inline cmd/manager/v1alpha3/main.go]: maybe we can add pkg/db/v1alpha3/common/const.go to define all costants including ENV name, db backend name and so on.

Comment (johnugeorge) [reply cmd/manager/v1alpha3/main.go]: Done

**Pull request merged**